full+min version instead of sourcemaps, main key, autoplay feature an…

galkahana opened the pull request:
Some things that i did:

1. gulp items:
   1. added "main" - this should help "bowerfiles" gulp task for users.
   2. gulp had shrinkwrap part but didn't have the npm resource. added code, added npm resource
   3. sourcemaps didn't work well, couldn't place breakpoints. so instead of min with sourcemap, i create minified and full versions, so that one can pick which to debug and which to use in production. 
2. player
  1. added autoplay option by toggling players. we have player that autoplays, and player that does not. that's the simplest and most stable implementation that i found so far.

tomersela commented on gulpfile.js:
```diff
@@ -57,18 +58,25 @@ gulp.task('clean:maps', function() {
     // compress and copy angular
-    gulp.src([config.srcFolder + '/**/*.js'])
-        .pipe(sourcemaps.init())
+    gulp.src([config.srcFolder + '/*.js'])
```
:+1:

tomersela commented on src/vidible-player.js:
```diff
@@ -11,9 +11,14 @@
-            function loadScriptByVideoId(videoId, callback) {
-                var vidibleScriptUrl = 'http://delivery.vidible.tv/jsonp/pid=55c8aae9e4b0ca68372fb553/vid=' +
-                    videoId + '/55af9dcae4b02944c03a2eee.js';
+            var kAutoplayVidiblePlayer = '55c8aae9e4b0ca68372fb553';
```
We should make player ID's something a that can be configured outside of the directive.
I used a static one as you can see, but there's a task for adding configuration to it.

tomersela replied:
Thought about it more. We'll leave it so for the moment. And I'll add a configuration option later that will include:
- playerIdWhenAutoPlayIsEnabled
- playerIdWhenAutoPlayIsDisabled
- vidibleId

tomersela commented on src/vidible-player.js:
```diff
@@ -44,6 +49,11 @@
                 }
             }
 
+            function playerForAutoplay(inAutoplay)
```
change to getAppropriatePlayer(isAutoPlayEnabled)

tomersela commented on gulpfile.js:
```diff
@@ -9,6 +9,7 @@ var gulp = require('gulp'),
     rename = require('gulp-rename'),
     minifyCss = require('gulp-minify-css'),
     sourcemaps = require('gulp-sourcemaps'),
+    shrinkwrap = require('gulp-shrinkwrap'),
```
:kissing_heart:

tomersela commented:
Looks great! thanks

tomersela commented on src/vidible-player.js:
```diff
@@ -206,7 +220,13 @@
+                            vidibleQueueLoader.queueForProcessing(
+                                {
+                                    videoId:videoId,
+                                    autoplay:scope.autoplay
```
Let's keep this method call simple - queueForProcessing(videoId, isAutoplay, element, initPlayer)
and take care for what to put it the queue in the implementation of queueForProcessing

tomersela commented:
Looks great! thanks

galkahana commented:
Added commit. includes the following:
1. Added replay option. simply recreates the player. other options (seekTo+Play) triggered vidible errors, so just didn't deal with it
2. isMuted and Mute status events are now more strict in terms of undefined/null cases, to better behave when transitioning between players or on the first value set. now we can get the event on true changes only. this is something i needed for the spree usage
3. added loaded and timeupdate events so i can track video loaded and playing progress

tomersela commented on src/vidible-player.js:
```diff
@@ -179,7 +211,11 @@
+                                            return undefined;
+                                        else
+                                            return vidiblePlayer.getPlayerInfo().volume === 0;
                                     } catch (e) {
```
Remove the try and catch. if they want to dirt the console let them, I don't want to swallow exceptions in case they will change the implementation in the future.

tomersela replied:
Is there a situation where vidiblePlayer.getPlayerInfo().volume throws undefined?
If yes we should check using if statements.

tomersela commented on src/vidible-player.js:
```diff
@@ -145,10 +168,14 @@
-                                if (currentState !== newState) {
-                                    var eventName = eventPrefix + (newState ? 'muted' : 'unmuted');
-                                    broadcastEvent(eventName, player, {muted: newState});
+                                if(currentState !== undefined) {
```
if(currentState !== undefined && currentState !== newState) {

tomersela commented on src/vidible-player.js:
```diff
@@ -166,6 +193,11 @@
                                     vidiblePlayer.play();
                                 },
+                                replay: function() {
+                                    // simple seek and play is buggy (throws exceptions from vidible), so i'm just recreating the player
```
:+1:

tomersela commented on src/vidible-player.js:
```diff
@@ -103,6 +119,10 @@
                             return eventPrefix + 'paused';
                         case vidible.VIDEO_PLAY:
                             return eventPrefix + 'playing';
+                        case vidible.VIDEO_DATA_LOADED:
```
Cool
